dashboard/head.py

```diff
@@ -122,8 +130,21 @@ def _load_modules(self):
 
     async def run(self):
         # Create an aioredis client for all modules.
-        self.aioredis_client = await aioredis.create_redis_pool(
-            address=self.redis_address, password=self.redis_password)
+        try:
+            self.aioredis_client = await dashboard_utils.get_aioredis_client(
+                self.redis_address, self.redis_password,
+                dashboard_consts.CONNECT_REDIS_INTERNAL_SECONDS,
+                dashboard_consts.RETRY_REDIS_CONNECTION_TIMES)
+        except (socket.gaierror, ConnectionError):
+            logger.error(
+                "Dashboard head suicide, "
+                "Failed to connect to redis at %s", self.redis_address)
+            sys.exit(-1)
```

Comment:
When the head is killed, All the agents will still be alive?

Reply:
Yes, agents are alive with raylet.

Reply:
Who's in charge of restarting the head dashboard in this case?

Reply:
The one who restart the `GCS` has the responsibility to restart the `dashboard`. Maybe it's better to fork `dashboard` by `GCS`?